sunpy/io/cdf.py

```diff
@@ -63,11 +64,18 @@ def read_cdf(fname):
 
             data = cdf.varget(var_key)
             # Get units
-            unit = attrs['UNITS']
-            if unit in ['None', '', 'unitless']:
-                unit = u.dimensionless_unscaled
-            else:
-                unit = u.Unit(unit)
+            unit_str = attrs['UNITS']
+            try:
+                unit = u.Unit(unit_str)
+            except ValueError as e:
+                if unit_str in _known_units:
+                    unit = _known_units[unit_str]
+                else:
+                    raise warn_user(f'astropy did not recognize units of "{unit_str}". '
```

Comment:
```suggestion
                    warn_user(f'astropy did not recognize units of "{unit_str}". '
```
I dont think you need `raise` here - it causes an error if its there

Reply:
Woops, thanks!